pyscript.core/esm/runtime/_python.js

```diff
@@ -0,0 +1,27 @@
+import { clean, writeFile as writeFileUtil } from "./_utils.js";
+
+// REQUIRES INTEGRATION TEST
+/* c8 ignore start */
+export const run = (runtime, code) => runtime.runPython(clean(code));
+
+export const runAsync = (runtime, code) => runtime.runPythonAsync(clean(code));
+
+export function runEvent(runtime, code, key) {
+    code = `import js;event=js.__events.get(${key});${code}`;
+    return this.run(runtime, code);
+}
+
+const worker = (method) =>
+    function (runtime, code, xworker) {
+        code = `from js import xworker;${code}`;
+        globalThis.xworker = xworker;
+        return this[method](runtime, code);
+    };
+
+export const runWorker = worker("run");
+
+export const runWorkerAsync = worker("runAsync");
```

Comment:
It seems like the main thing that `runWorker` does is bind the extra `xworker` variable. I wonder whether it would be a better design to add an extra `context` argument to `run` and `runAsync` and pass `{ xworker }` as the context argument. The current design seems like it couples the runtime logic too tightly with the worker support. Then depending on the runtime, different methods could be used to bind the context.

Reply:
the worker is tightly coupled with `xworker` on purpose as the whole thing wouldn't work otherwise (onmessage, onerror, etc) and this is consistent across all runtimes already but I think I am not following what you are saying here.

Reply:
I'm saying that all `runWorker` does is call run with extra context so it would be better for the runtime to define an API `runWithExtraContext` instead.

Reply:
`xworker` is mandatory to make anything work in here ... if you have a better proposal and better DX/UX in mind, please file a PR :+1: